src/hb-ot-layout-common.hh

```diff
@@ -2070,6 +2072,12 @@ struct HintingDevice
     return_trace (c->check_struct (this) && c->check_range (this, this->get_size ()));
   }
 
+  HintingDevice* copy (hb_serialize_context_t *c) const
+  {
+    TRACE_SERIALIZE (this);
+    return_trace (c->embed<HintingDevice> (this));
```

Comment:
Do you need to specify the type?  It should be deduces from `this` automatically, no?  I think that's how we do in other places.